### PR TITLE
 feat(sdk): support `display_name` and `description` in `@dsl.pipeline` decorator

### PR DESCRIPTION
**Pull request by connor-mccarthy**
**Description of your changes:**

The `@dsl.pipeline` decorator now accepts `display_name` and `description` arguments. These are compiled to IR YAML.

```python
from kfp import dsl

@dsl.pipeline(
    display_name='Human readable name',
    description='My description. Decorator argument takes precedence over docstring',
)
def my_pipeline():
    """Will use docstring description when decorator description is not provided.
         In this case, description is provided to decorator is docstring is unused.
    """
    ...
```

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more 
about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: sdk/python/test_data/pipelines/pipeline_with_nested_conditions_yaml.yaml [hunk 1]
```diff
@@ -302,6 +302,7 @@ deploymentSpec:
         - '{{$.inputs.parameters[''msg'']}}'
         image: python:alpine3.6
 pipelineInfo:
+  description: Shows how to use dsl.Condition().
   name: conditional-execution-pipeline
```
Comment (chensun) [inline sdk/python/test_data/pipelines/pipeline_with_nested_conditions_yaml.yaml:306]: Can we at least have one pipeline that sets both `name` and `display_name` to make sure we are not mixing the two?

Comment (connor-mccarthy) [reply sdk/python/test_data/pipelines/pipeline_with_nested_conditions_yaml.yaml:307]: Yes. Added a unit test and a compilation test for this case.

**Review by chensun — APPROVED** [review]
/lgtm
/approve

**Pull request merged**